src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -53,12 +57,28 @@ public override async Task OnConnectedAsync(HubConnectionContext connection)
                     hubActivator.Release(hub);
                 }
             }
+            finally
```

Comment:
Seems like alot of rinse repeat with the dispose pattern, could scope to method and inline.

Reply:
If you mean make a helper method that takes a lambda then that would have worse performance than just doing it inline.

Reply:
Extension method is sweet